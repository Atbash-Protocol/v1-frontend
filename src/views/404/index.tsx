import { useTranslation } from 'react-i18next';
import { Box, Typography } from '@mui/material';
import { theme } from 'constants/theme';

function NotFound() {
    const { t } = useTranslation();

    return (
        <Box
            sx={{
                justifyContent: 'center',
                alignItems: 'center',
                display: 'flex',
                height: '100%',
            }}
        >
            <Typography variant="h1" sx={{ textAlign: 'center', fontWeight: '600', color: theme.palette.secondary.main }}>
                {t('PageNotFound')}
            </Typography>
        </Box>
    );
}

export default NotFound;
