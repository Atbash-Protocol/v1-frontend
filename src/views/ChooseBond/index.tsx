import { useDispatch, useSelector } from "react-redux";
import { Box, Typography, Grid } from "@mui/material";
import { BondMint } from "./BondRow";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { formatUSD } from "../../helpers";
// import "./choosebond.scss";
import { IReduxState } from "../../store/slices/state.interface";

import { useTranslation } from "react-i18next";
import { selectAllBonds } from "store/modules/bonds/bonds.selector";
import { selectDAIPrice } from "store/modules/markets/markets.selectors";
import { useEffect } from "react";
import { calcBondDetails, getTreasuryBalance } from "store/modules/bonds/bonds.thunks";
import { useWeb3Context } from "hooks/web3";
import { theme } from "constants/theme";
import { MenuMetric } from "components/Metrics/MenuMetric";
import { BCard } from "components/BCard";

const BondHeader = () => {
    const { t } = useTranslation();

    return (
        <Grid
            container
            sx={{
                [theme.breakpoints.up("xs")]: {
                    display: "none",
                },
                [theme.breakpoints.up("sm")]: {
                    display: "inline-flex",
                },
                color: theme.palette.secondary.main,
            }}
        >
            <Grid item sm={1} />
            <Grid item sm={2}>
                <Typography variant="h6">{t("bond:Mint")}</Typography>
            </Grid>
            <Grid item sm={2}>
                <Typography variant="h6">{t("Price")}</Typography>
            </Grid>
            <Grid item sm={2}>
                <Typography variant="h6">{t("ROI")}</Typography>
            </Grid>
            <Grid item sm={2}>
                <Typography variant="h6">{t("bond:Purchased")}</Typography>
            </Grid>
            <Grid item sm={2} />
        </Grid>
    );
};

function ChooseBond() {
    const { t } = useTranslation();
    const { chainID } = useWeb3Context();
    const dispatch = useDispatch();

    const { activeBonds, inactiveBonds } = useSelector(selectAllBonds);

    const isSmallScreen = useMediaQuery("(max-width: 733px)"); // change to breakpoint query

    const marketPrice = useSelector<IReduxState, number | null>(selectDAIPrice);

    const treasuryBalance = useSelector<IReduxState, number | null>(state => state.bonds.treasuryBalance);

    const isAppLoading = !marketPrice || !treasuryBalance;

    console.log(marketPrice, isAppLoading, treasuryBalance, activeBonds);

    useEffect(() => {
        if (!isAppLoading) {
            dispatch(calcBondDetails({ bond: activeBonds[0], value: 0 }));
        }
    }, [isAppLoading]);

    return (
        <>
            <BCard title={t("bond:MintTitle")} zoom={true}>
                <Box>
                    <Grid container item xs={12} spacing={2} mb={4}>
                        <Grid item xs={12} sm={6}>
                            <MenuMetric metricKey={t("TreasuryBalance")} value={treasuryBalance ? formatUSD(treasuryBalance) : null} />
                        </Grid>

                        <Grid item xs={12} sm={6}>
                            <MenuMetric metricKey={t("BASHPrice")} value={marketPrice ? formatUSD(marketPrice, 2) : null} />
                        </Grid>
                    </Grid>
                    <Grid container item>
                        <BondHeader />

                        {activeBonds.map(bond => (
                            <BondMint key={bond.ID} bondID={bond.ID} />
                        ))}
                    </Grid>
                    )
                </Box>
            </BCard>

            <BCard title={t("bond:MintInactiveTitle")} zoom={true} className="BondList__card">
                <Box>
                    <Grid container item>
                        <BondHeader />
                        {inactiveBonds.map(bond => (
                            <BondMint key={bond.ID} bondID={bond.ID} />
                        ))}
                    </Grid>
                    )
                </Box>
            </BCard>
        </>
    );
}

export default ChooseBond;
