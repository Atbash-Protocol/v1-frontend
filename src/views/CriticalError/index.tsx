import { Box, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';

const CritialError = () => {
    const { t } = useTranslation('common');

    return (
        <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <Typography variant="h1">
                <>{t('errors.critical')}</>
            </Typography>
        </Box>
    );
};

export default CritialError;
