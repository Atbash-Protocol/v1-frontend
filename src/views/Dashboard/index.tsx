import { useSelector } from 'react-redux';
import { Zoom } from '@material-ui/core';
import { formatUSD, formatNumber } from 'helpers/price-units';
import { IReduxState } from 'store/slices/state.interface';
import { IAppSlice } from 'store/slices/app-slice';

import { useTranslation } from 'react-i18next';
import Grid from '@mui/material/Grid';
import { Box } from '@mui/material';
import { theme } from 'constants/theme';
import { Metric } from './components/Metric';

import './dashboard.scss';

function Dashboard() {
    const { t } = useTranslation();

    const { loading: isAppLoading, ...app } = useSelector<IReduxState, IAppSlice>(state => state.app);

    const formattedAPY = formatNumber(app.stakingAPY * 100, 1);
    const trimmedStakingAPY = formattedAPY.length > 16 ? '> 1,000,000' : formattedAPY;

    const DashboardItems = [
        { name: 'RiskFreeValue', value: formatUSD(app.rfv) },
        { name: 'RiskFreeValuewsBASH', value: formatUSD(app.rfv * Number(app.currentIndex)) },
        { name: 'BashPrice', value: formatNumber(app.marketPrice, 2) },
        { name: 'wsBASHPrice', value: formatNumber(app.marketPrice * Number(app.currentIndex), 2) },
        { name: 'MarketCap', value: app.marketCap },
        { name: 'TVL', value: app.stakingTVL },
        { name: 'APY', value: `${trimmedStakingAPY} %` },
        { name: 'CurrentIndex', value: `${formatNumber(Number(app.currentIndex), 2)} BASH` },
        { name: 'TreasuryBalance', value: formatUSD(app.treasuryBalance, 0) },
        { name: 'Runway', value: `${formatNumber(Number(app.runway), 1)} Days` },
    ];

    return (
        <Box>
            <Zoom in={true}>
                <Grid container spacing={6} sx={{ p: 2 }} justifyContent="space-around">
                    {DashboardItems.map(metric => (
                        <Grid key={`dashboard-item-${metric.name}`} item lg={6} md={6} sm={6} xs={12}>
                            <Box
                                className="Dashboard__box__item"
                                sx={{
                                    backgroundColor: theme.palette.cardBackground.main,
                                    backdropFilter: 'blur(100px)',
                                    borderRadius: '.5rem',
                                    color: theme.palette.secondary.main,
                                    px: theme.spacing(2),
                                    py: theme.spacing(4),
                                    textAlign: 'center',
                                }}
                            >
                                <Metric name={t(metric.name)} value={metric.value} loading={isAppLoading} />
                            </Box>
                        </Grid>
                    ))}
                </Grid>
            </Zoom>
        </Box>
    );
}

export default Dashboard;
