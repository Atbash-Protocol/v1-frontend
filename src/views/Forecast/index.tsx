import { useState } from 'react';
import { useSelector } from 'react-redux';
import { Box, Grid, InputAdornment, OutlinedInput, Zoom } from '@material-ui/core';
import Slider from '@mui/material/Slider';
import { trim } from '../../helpers';
import './Forecast.scss';
import { useWeb3Context } from '../../hooks';
import { Skeleton } from '@material-ui/lab';
import { IReduxState } from '../../store/slices/state.interface';
import { fCurrency, fShortenNumber } from '../../helpers/formatNumers';

import { IAppSlice } from '../../store/slices/app-slice';

import { useTranslation } from 'react-i18next';

function Stake() {
    const { t } = useTranslation();

    const { address, connect } = useWeb3Context();

    const [sBASHQuantity, setsBASHQuantity] = useState<string>('1');
    const [rewardYield, setRewardYield] = useState<string>('1.5');
    const [price, setPrice] = useState<string>('1200');
    const [futurePrice, setFuturePrice] = useState<string>('10000');
    const [duration, setDuration] = useState<number>(30);

    const isAppLoading = useSelector<IReduxState, boolean>(state => state.app.loading);
    const sBASHBalance = useSelector<IReduxState, string>(state => {
        return state.account.balances && state.account.balances.sBASH;
    });
    const stakingRebase = useSelector<IReduxState, number>(state => {
        return state.app.stakingRebase;
    });

    const app = useSelector<IReduxState, IAppSlice>(state => state.app);

    const setsBASHToMax = () => {
        setsBASHQuantity(sBASHBalance);
    };

    const trimmedsBASHBalance = trim(Number(sBASHBalance), 6);
    const stakingRebasePercentage = trim(stakingRebase * 100, 4);

    const setYeildCurrent = () => {
        setRewardYield(stakingRebasePercentage);
    };
    const setCurrentPrice = () => {
        setPrice(trim(app.marketPrice, 2));
    };
    const setCurrentPriceToFuture = () => {
        setFuturePrice(trim(app.marketPrice, 2));
    };

    function preventHorizontalKeyboardNavigation(event: React.KeyboardEvent) {
        if (event.key === 'ArrowLeft' || event.key === 'ArrowRight') {
            event.preventDefault();
        }
    }
    const handleSliderChange = (event: Event, newValue: number) => {
        setDuration(newValue);
    };

    // Forecast calc
    const epochs_per_day = 3;

    const roi_day = (1 + parseFloat(rewardYield) / 100) ** (duration * epochs_per_day) - 1;
    const gains = parseFloat(sBASHQuantity) * roi_day;
    const lamboPrice = 222004;

    const currentWealth = parseFloat(sBASHQuantity) * app?.marketPrice;
    const futureWealth = (parseFloat(sBASHQuantity) + gains) * parseFloat(futurePrice);
    const nbLambo = Math.round(futureWealth / lamboPrice);
    const investment = parseFloat(price) * parseFloat(sBASHQuantity);

    return (
        <div className="stake-view">
            <Zoom in={true}>
                <div className="Forecast-card">
                    <Grid className="Forecast-card-grid" container direction="column" spacing={2}>
                        <Grid item>
                            <div className="Forecast-card-header">
                                <p className="Forecast-card-header-title">{t('globe:ForecastTitle')}</p>
                                <p className="Forecast-card-header-subtitle">{t('globe:EstimateYourReturns')}</p>
                                <p className="Forecast-card-header-disclaimer">{t('globe:ForecastWarning')}</p>
                            </div>
                        </Grid>

                        <Grid item>
                            <div className="Forecast-card-metrics">
                                <Grid container spacing={2}>
                                    <Grid item xs={12} sm={4} md={4} lg={4}>
                                        <div className="Forecast-card-apy">
                                            <p className="Forecast-card-metrics-title">{t('globe:CurrentBASHPrice')}</p>
                                            <p className="Forecast-card-metrics-value">{isAppLoading ? <Skeleton width="100px" /> : `$${trim(app.marketPrice, 2)}`}</p>
                                        </div>
                                    </Grid>

                                    <Grid item xs={6} sm={4} md={4} lg={4}>
                                        <div className="Forecast-card-tvl">
                                            <p className="Forecast-card-metrics-title">{t('globe:CurrentRewardYield')}</p>
                                            <p className="Forecast-card-metrics-value">{isAppLoading ? <Skeleton width="80px" /> : <>{stakingRebasePercentage}%</>}</p>
                                        </div>
                                    </Grid>

                                    <Grid item xs={6} sm={4} md={4} lg={4}>
                                        <div className="Forecast-card-index">
                                            <p className="Forecast-card-metrics-title">{t('globe:YourStakedBASHBalance')}</p>
                                            <p className="Forecast-card-metrics-value">{isAppLoading ? <Skeleton width="150px" /> : <>{trimmedsBASHBalance} sBASH</>}</p>
                                        </div>
                                    </Grid>
                                </Grid>
                            </div>
                        </Grid>

                        <div className="Forecast-card-area">
                            {!address && (
                                <div className="Forecast-card-wallet-notification">
                                    <div className="Forecast-card-wallet-connect-btn" onClick={connect}>
                                        <p>{t('ConnectWallet')}</p>
                                    </div>
                                    <p className="Forecast-card-wallet-desc-text">{t('globe:ConnectYourWalletToUseForecast')}</p>
                                </div>
                            )}
                            {address && (
                                <div className="Forecast-card-action-area">
                                    <Grid className="Forecast-card-grid" container spacing={1}>
                                        <Grid item sm={5} className="Forecast-entry">
                                            <div className="Forecast-card-action-row">
                                                <p className="Forecast-card-action-label">{t('globe:StakedSBAmount')}</p>
                                                <OutlinedInput
                                                    type="number"
                                                    placeholder={t('globe:EnterStakedSBAmount')}
                                                    className="Forecast-card-action-input"
                                                    value={sBASHQuantity}
                                                    onChange={e => setsBASHQuantity(e.target.value)}
                                                    labelWidth={0}
                                                    endAdornment={
                                                        <InputAdornment position="end">
                                                            <div onClick={setsBASHToMax} className="Forecast-card-action-input-btn">
                                                                <p>{t('Max')}</p>
                                                            </div>
                                                        </InputAdornment>
                                                    }
                                                />
                                            </div>
                                            <div className="Forecast-card-action-row">
                                                <p className="Forecast-card-action-label">{t('globe:RewardYieldPercent')}</p>
                                                <OutlinedInput
                                                    type="number"
                                                    placeholder={t('globe:EnterRewardYieldPercent')}
                                                    className="Forecast-card-action-input"
                                                    value={rewardYield}
                                                    onChange={e => setRewardYield(e.target.value)}
                                                    labelWidth={0}
                                                    endAdornment={
                                                        <InputAdornment position="end">
                                                            <div onClick={setYeildCurrent} className="Forecast-card-action-input-btn">
                                                                <p>{t('globe:Current')}</p>
                                                            </div>
                                                        </InputAdornment>
                                                    }
                                                />
                                            </div>
                                            <div className="Forecast-card-action-row">
                                                <p className="Forecast-card-action-label">{t('globe:BASHPriceAtPurchase')}</p>
                                                <OutlinedInput
                                                    type="number"
                                                    placeholder={t('globe:EnterBuyPrice')}
                                                    className="Forecast-card-action-input"
                                                    value={price}
                                                    onChange={e => setPrice(e.target.value)}
                                                    labelWidth={0}
                                                    endAdornment={
                                                        <InputAdornment position="end">
                                                            <div onClick={setCurrentPrice} className="Forecast-card-action-input-btn">
                                                                <p>{t('globe:Current')}</p>
                                                            </div>
                                                        </InputAdornment>
                                                    }
                                                />
                                            </div>
                                            <div className="Forecast-card-action-row">
                                                <p className="Forecast-card-action-label">{t('globe:FutureBASHMarketPrice')}</p>
                                                <OutlinedInput
                                                    type="number"
                                                    placeholder={t('globe:EnterFuturePrice')}
                                                    className="Forecast-card-action-input"
                                                    value={futurePrice}
                                                    onChange={e => setFuturePrice(e.target.value)}
                                                    labelWidth={0}
                                                    endAdornment={
                                                        <InputAdornment position="end">
                                                            <div onClick={setCurrentPriceToFuture} className="Forecast-card-action-input-btn">
                                                                <p>{t('globe:Current')}</p>
                                                            </div>
                                                        </InputAdornment>
                                                    }
                                                />
                                            </div>
                                        </Grid>
                                        <Grid item className="slider-container">
                                            <Box className="slider-inner">
                                                <Box className="Forecast-days">
                                                    <p className="data-row-value">
                                                        {duration}
                                                        <br />
                                                        {t('day', { count: duration })}
                                                    </p>
                                                </Box>
                                                <Slider
                                                    sx={{
                                                        '& input[type="range"]': {
                                                            WebkitAppearance: 'slider-vertical',
                                                        },
                                                        display: { xs: 'none', sm: 'block' },
                                                    }}
                                                    step={1}
                                                    marks
                                                    min={1}
                                                    max={365}
                                                    orientation="vertical"
                                                    value={duration}
                                                    //@ts-ignore
                                                    onChange={handleSliderChange}
                                                    aria-label={t('globe:Duration')}
                                                    onKeyDown={preventHorizontalKeyboardNavigation}
                                                />
                                                <Slider
                                                    sx={{
                                                        '& input[type="range"]': {
                                                            WebkitAppearance: 'slider-horizontal',
                                                        },
                                                        display: { xs: 'block', sm: 'none' },
                                                        width: 1,
                                                    }}
                                                    step={1}
                                                    marks
                                                    min={1}
                                                    max={365}
                                                    orientation="horizontal"
                                                    value={duration}
                                                    //@ts-ignore
                                                    onChange={handleSliderChange}
                                                    aria-label={t('globe:Duration')}
                                                    onKeyDown={preventHorizontalKeyboardNavigation}
                                                />
                                            </Box>
                                        </Grid>

                                        <Grid item sm={5} className="Forecast-data">
                                            <div className="data-row">
                                                <p className="data-row-name">{t('globe:YourInitialInvestment')}</p>
                                                <p className="data-row-value">{fCurrency(investment)}</p>
                                            </div>
                                            <div className="data-row">
                                                <p className="data-row-name">{t('globe:CurrentWealth')}</p>
                                                <p className="data-row-value">{fCurrency(currentWealth)}</p>
                                            </div>
                                            <div className="data-row">
                                                <p className="data-row-name">{t('globe:BASHRewardsEstimation')}</p>
                                                <p className="data-row-value">{fShortenNumber(gains)} BASH</p>
                                            </div>
                                            <div className="data-row">
                                                <p className="data-row-name">{t('globe:PotentialReturn')}</p>
                                                <p className="data-row-value">{fCurrency(futureWealth)}</p>
                                            </div>
                                            <div className="data-row">
                                                <p className="data-row-name">
                                                    {t('globe:PotentialNumberLambo')}
                                                    <span>{t('globe:LamboHelpText')}</span>
                                                </p>
                                                <p className="data-row-value">{nbLambo}</p>
                                            </div>
                                        </Grid>
                                    </Grid>
                                </div>
                            )}
                        </div>
                    </Grid>
                </div>
            </Zoom>
        </div>
    );
}

export default Stake;
