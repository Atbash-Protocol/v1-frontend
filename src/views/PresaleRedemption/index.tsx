import { useState, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Grid, InputAdornment, OutlinedInput, Zoom, Link } from '@material-ui/core';
import { trim } from '../../helpers';
import { changeRedeem, changeRedeemApproval } from '../../store/slices/redeem-thunk';
import { useWeb3Context } from '../../hooks';
import { IPendingTxn, isPendingTxn, txnButtonText } from '../../store/slices/pending-txns-slice';
import { Skeleton } from '@material-ui/lab';
import { IReduxState } from '../../store/slices/state.interface';
import { messages } from '../../constants/messages';
import { warning } from '../../store/slices/messages-slice';
import { IAppSlice, loadAppDetails } from '../../store/slices/app-slice';

import { useTranslation } from 'react-i18next';

import './presale-redemption.scss';

function PresaleRedemption() {
    const { t } = useTranslation();

    const dispatch = useDispatch();
    const { provider, address, connect, chainID, checkWrongNetwork } = useWeb3Context();
    const app = useSelector<IReduxState, IAppSlice>(state => state.app);

    const [quantity, setQuantity] = useState<string>('');

    const isAppLoading = useSelector<IReduxState, boolean>(state => state.app.loading);
    const BashBalance = useSelector<IReduxState, string>(state => {
        return state.account.balances && state.account.balances.BASH;
    });
    const abashBalance = useSelector<IReduxState, string>(state => {
        return state.account.balances && state.account.balances.aBash;
    });
    const redeemAllowance = useSelector<IReduxState, number>(state => {
        return state.account.redeeming?.aBash;
    });
    const pendingTransactions = useSelector<IReduxState, IPendingTxn[]>(state => {
        return state.pendingTransactions;
    });
    const redeemableAbash = useSelector<IReduxState, number>(state => {
        return state.app.redeemableAbash;
    });

    const setMax = () => {
        setQuantity(abashBalance);
    };

    const onSeekApproval = async (token: string) => {
        if (await checkWrongNetwork()) return;

        await dispatch(changeRedeemApproval({ address, token, provider, networkID: chainID }));
    };

    const onChangeRedeem = async (action: string) => {
        if (await checkWrongNetwork()) return;
        if (quantity === '' || parseFloat(quantity) === 0) {
            dispatch(warning({ text: messages.before_redeem }));
        } else {
            await dispatch(changeRedeem({ address, action, value: String(quantity), provider, networkID: chainID }));
            setQuantity('');
            let loadProvider = provider;
            dispatch(loadAppDetails({ networkID: chainID, provider: loadProvider }));
        }
    };

    const hasAllowance = useCallback(
        token => {
            return redeemAllowance;
        },
        [redeemAllowance],
    );

    const userRedeemableAbash = new Intl.NumberFormat('en-US', {
        maximumFractionDigits: 8,
        minimumFractionDigits: 2,
    }).format(Number(abashBalance));

    return (
        <div className="stake-view">
            <Zoom in={true}>
                <div className="stake-card">
                    <Grid className="stake-card-grid" container direction="column" spacing={2}>
                        <Grid item>
                            <div className="stake-card-header">
                                <p className="stake-card-header-title">{t('redeem:RedeemTitle')}</p>
                            </div>
                        </Grid>

                        <div className="stake-card-area">
                            {!address && (
                                <div className="stake-card-wallet-notification">
                                    <div className="stake-card-wallet-connect-btn" onClick={connect}>
                                        <p>{t('ConnectWallet')}</p>
                                    </div>
                                    <p className="stake-card-wallet-desc-text">{t('redeem:ConnectYourWalletToStake')}</p>
                                </div>
                            )}
                            {address && (
                                <>
                                    <div className="stake-card-action-area">
                                        <div className="stake-card-action-row">
                                            <OutlinedInput
                                                type="number"
                                                placeholder={t('Amount')}
                                                className="stake-card-action-input"
                                                value={quantity}
                                                onChange={e => setQuantity(e.target.value)}
                                                labelWidth={0}
                                                endAdornment={
                                                    <InputAdornment position="end">
                                                        <div onClick={setMax} className="stake-card-action-input-btn">
                                                            <p>{t('Max')}</p>
                                                        </div>
                                                    </InputAdornment>
                                                }
                                            />

                                            <div className="stake-card-tab-panel">
                                                {address && hasAllowance('aBash') ? (
                                                    <div
                                                        className="stake-card-tab-panel-btn"
                                                        onClick={() => {
                                                            if (isPendingTxn(pendingTransactions, 'redeem')) return;
                                                            onChangeRedeem('redeem');
                                                        }}
                                                    >
                                                        <p>{txnButtonText(pendingTransactions, 'redeeming', t('redeem:Redeem'))}</p>
                                                    </div>
                                                ) : (
                                                    <div
                                                        className="stake-card-tab-panel-btn"
                                                        onClick={() => {
                                                            if (isPendingTxn(pendingTransactions, 'approve_redeem')) return;
                                                            onSeekApproval('aBash');
                                                        }}
                                                    >
                                                        <p>{txnButtonText(pendingTransactions, 'approve_redeem', t('Approve'))}</p>
                                                    </div>
                                                )}
                                            </div>
                                        </div>

                                        <div className="stake-card-action-help-text">{address && !hasAllowance('aBash') && <p>{t('redeem:ApproveNote')}</p>}</div>
                                    </div>

                                    <div className="stake-user-data">
                                        <div className="data-row">
                                            <p className="data-row-name">Your Abash avaialble to Redeem</p>
                                            <p className="data-row-value">{isAppLoading ? <Skeleton width="80px" /> : <>{userRedeemableAbash}</>}</p>
                                        </div>
                                        <div className="data-row">
                                            <p className="data-row-name">Your BASH Balance</p>
                                            <p className="data-row-value">{isAppLoading ? <Skeleton width="80px" /> : <>{trim(Number(BashBalance), 4)} BASH</>}</p>
                                        </div>
                                        <div className="data-row">
                                            <p className="data-row-name">{t('Total Presale Still Redeemable')}</p>
                                            <p className="data-row-value">{isAppLoading ? <Skeleton width="80px" /> : <>{redeemableAbash}</>}</p>
                                        </div>
                                    </div>
                                </>
                            )}
                        </div>
                    </Grid>
                </div>
            </Zoom>
        </div>
    );
}

export default PresaleRedemption;
