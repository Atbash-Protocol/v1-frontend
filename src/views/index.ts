export { default as Bond } from "./Bond";
export { default as ChooseBond } from "./ChooseBond";
export { default as Stake } from "./Staking";
export { default as Redeem } from "./Redeem";
export { default as Forecast } from "./Forecast";
export { default as Dashboard } from "./Dashboard";
export { default as NotFound } from "./404";
export { default as Wrap } from "./Wrap";
export { CritialError } from "./CriticalError";
