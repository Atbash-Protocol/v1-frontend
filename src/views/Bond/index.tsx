import BondHeader from "./BondHeader";
import "./bond.scss";
import { IAllBondData } from "../../hooks/bonds";

import { Fade, Grid, Backdrop } from "@mui/material";
import bond from "helpers/bond";
import { useDispatch, useSelector } from "react-redux";
import { IReduxState } from "store/slices/state.interface";
import { useEffect, useState } from "react";
import { LPBond } from "lib/bonds/bond/lp-bond";
import { calcBondDetails, getTreasuryBalance } from "store/modules/bonds/bonds.thunks";
import { useWeb3Context } from "hooks/web3";
import { selectActiveBonds } from "store/modules/bonds/bonds.selector";
import { BondDataCard } from "views/ChooseBond/BondRow";

interface IBondProps {
    bond: IAllBondData;
}

function Bond() {
    // const { t } = useTranslation();

    const { chainID } = useWeb3Context();

    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(getTreasuryBalance(chainID));
    }, []);

    const bonds = useSelector<IReduxState, LPBond[]>(selectActiveBonds);
    useEffect(() => {
        if (bonds && bonds.length > 0) {
            dispatch(calcBondDetails({ bond: bonds[0], value: 10, chainID }));
        }
    }, [bonds]);

    const [slippage, setSlippage] = useState(0.5);
    const [recipientAddress, setRecipientAddress] = useState("address");

    // const [view, setView] = useState(bond.isActive ? 0 : 1);

    // const isBondLoading = useSelector<IReduxState, boolean>(state => state.bonding.loading ?? true);

    const onRecipientAddressChange = (value: any) => {
        return setRecipientAddress(value);
    };

    const onSlippageChange = (value: any) => {
        return setSlippage(value);
    };

    // useEffect(() => {
    //     if (address) setRecipientAddress(address);
    // }, [provider, address]);

    // const changeView = (newView: number) => () => {
    //     setView(newView);

    // return (
    //     <Fade in={true} mountOnEnter unmountOnExit>
    //         <Grid className="bond-view">
    //             <Backdrop open={true}>
    //                 <Fade in={true}>
    //                     <div className="bond-card">
    //                         <BondHeader
    //                             bond={bond}
    //                             slippage={slippage}
    //                             recipientAddress={recipientAddress}
    //                             onSlippageChange={onSlippageChange}
    //                             onRecipientAddressChange={onRecipientAddressChange}
    //                         />
    //                         {/* @ts-ignore */}
    //                         <Box direction="row" className="bond-price-data-row">
    //                             <div className="bond-price-data">
    //                                 <p className="bond-price-data-title">{t("bond:MintPrice")}</p>
    //                                 <p className="bond-price-data-value">
    //                                     {isBondLoading ? <Skeleton /> : bond.isLP || bond.name === "wavax" ? `$${trim(bond.bondPrice, 2)}` : `${trim(bond.bondPrice, 2)} MIM`}
    //                                 </p>
    //                             </div>
    //                             <div className="bond-price-data">
    //                                 <p className="bond-price-data-title">{t("BASHPrice")}</p>
    //                                 <p className="bond-price-data-value">{isBondLoading ? <Skeleton /> : `$${trim(bond.marketPrice, 2)}`}</p>
    //                             </div>
    //                         </Box>

    //                         {bond.isActive && (
    //                             <div className="bond-one-table">
    //                                 <div className={classnames("bond-one-table-btn", { active: !view })} onClick={changeView(0)}>
    //                                     <p>{t("bond:Mint")}</p>
    //                                 </div>
    //                                 <div className={classnames("bond-one-table-btn", { active: view })} onClick={changeView(1)}>
    //                                     <p>{t("bond:Redeem")}</p>
    //                                 </div>
    //                             </div>
    //                         )}

    //                         {bond.isActive && (
    //                             <TabPanel value={view} index={0}>
    //                                 <BondPurchase bond={bond} slippage={slippage} recipientAddress={recipientAddress} />
    //                             </TabPanel>
    //                         )}

    //                         <TabPanel value={view} index={1}>
    //                             <BondRedeem bond={bond} />
    //                         </TabPanel>
    //                     </div>
    //                 </Fade>
    //             </Backdrop>
    //         </Grid>
    //     </Fade>
    // );

    // return (
    //     <Fade in={true} mountOnEnter unmountOnExit>
    //         <Grid className="bond-view">
    //             <Backdrop open={true}>
    //                 <Fade in={true}>
    //                     <div className="bond-card">
    //                         <BondHeader
    //                             bond={bond[0]}
    //                             slippage={slippage}
    //                             recipientAddress={recipientAddress}
    //                             onSlippageChange={onSlippageChange}
    //                             onRecipientAddressChange={onRecipientAddressChange}
    //                         />
    //                     </div>
    //                 </Fade>
    //             </Backdrop>
    //         </Grid>
    //     </Fade>
    // );

    return (
        <Grid container item spacing={2}>
            {bonds.map(bond => (
                <Grid item xs={12} key={bond.ID}>
                    <BondDataCard bondID={bond.ID} />
                </Grid>
            ))}
        </Grid>
    );
}

export default Bond;
