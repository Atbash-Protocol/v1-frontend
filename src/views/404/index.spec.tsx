import { render } from '@testing-library/react';

import { NotFound } from 'views';

describe('NotFound', () => {
    it('renders', () => {
        const { container } = render(<NotFound />);

        expect(container).toMatchSnapshot();
    });
});
