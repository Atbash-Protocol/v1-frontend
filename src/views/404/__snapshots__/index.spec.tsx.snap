// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders 1`] = `
<div>
  <div
    class="MuiBox-root css-9vzy08"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-lt69zl-MuiTypography-root"
    >
      PageNotFound
    </h1>
  </div>
</div>
`;
